src/Microsoft.Azure.WebJobs.Extensions.Kafka/Listeners/KafkaMetricsProvider.cs

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Confluent.Kafka;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Logging;
+using System;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Extensions.Kafka;
+using static Confluent.Kafka.ConfigPropertyNames;
+
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka
+{
+    public class KafkaMetricsProvider<TKey, TValue>
```

Comment:
Do we need this class to be public?

Reply:
It was made public to increase ease in writing tests. After changes made to tests to work with internal class, reverting the public class back to internal.